src/WebJobs.Extensions.NotificationHubs/Config/NotificationHubsConfiguration.cs

```diff
@@ -127,13 +128,13 @@ internal string ResolveHubName(string attributeHubName)
             // First, try the Attribute's string.
             if (!string.IsNullOrEmpty(attributeHubName))
             {
-                return attributeHubName;
+                return _nameResolver.ResolveWholeString(attributeHubName);
```

Comment:
Instead of using the name resolver directly, I wonder if applying the `[AppSetting]` or `[AutoResolve]` attribute to the HubName property in NotificationHubAttribute may make more sense.

AppSetting will resolve from app settings only, AutoResolve will resolve %% and {} within names to app settings and binding data respectively. In this case, I think that AppSetting makes the most sense.

https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs#L155

One potential issue is if `AppSetting` requires the property to exist - perhaps the attribute could have an 'optional' flag.